### DISCUSSION TITLE
Pdf has extra shapes inside the file

### DISCUSSION THREAD

**Discussion by louisschprs** [spatie/browsershot#619] [Q&A]
Disclaimer: I understand that browsershot was not meant to generate PPTX files nor generate pdfs to be converted to PPTX. My question is more about the shapes included in the PDF (I tested this with other pdf files generated by other means, this only happens with browsershot generated pdfs);

Im using browser shot to convert a laravel view into a PDF which works wonderful. I want to include a conversion to pptx on my site. After consulting with peers and doing research the best option seems to be to convert a pdf to pptx. To accomplish this I am making use of libreoffice command line to do the conversion. 

When opening the file in any presentation software It shows white shapes on the page. This looks like it is about the same size as the page. If i convert it to pptx its a solid white block. If i convert it to ppt it has some areas that are white and others that are transparent. 

This is from WPS office. ( This is a ppt file)
![image](https://user-images.githubusercontent.com/78848393/131472129-52fb6b8e-ac00-4f57-850b-9e60d7681c10.png)

This is from Libre office draw (This is the pdf file, libre office draw opens pdf files as if they were drawings, this then shows the shapes inside the pdf aswell, which gets carried over to the powerpoint in the conversion. Opening the pdf in a reader however doesnt show the shapes. 
![image](https://user-images.githubusercontent.com/78848393/131473771-479acf44-727f-43f4-8406-517776da252f.png)
![image](https://user-images.githubusercontent.com/78848393/131473874-85669205-7afa-42b4-8cc9-55e602bdfe77.png)

It seems that It creates 2 shapes on each page.
![image](https://user-images.githubusercontent.com/78848393/131474098-5969fe23-c17a-4b5d-8163-6489874422ee.png)

I have tried to generate pdfs with every imaginable combination of settings and commands on browsershot. 

I tried with changing the page size:
```
Browsershot::html($someHtml)
   ->paperSize($width, $height)
   ->save('example.pdf');
```
I tried Changing it to landscape.
I tried converting the pdf from html instead by grabbing the html object and then saving it. 

I was wondering is this a bug? Is it browsershot? Is it puppeteer ? Is it Chrome?. Is their a way in browser shot to remove this?

**Comment by Lednerb** [comment]
I've had a similar but different issue that was related with not loaded libraries at snapshot / pdf generating time.

You can try to chain the following methods, maybe it will help:

```php
->setOption('args', ['--disable-web-security']) // maybe fixes issues related to cors, etc
->waitUntilNetworkIdle() // fixes issues with rendering
```
